src/methods/environment.js

```diff
@@ -459,33 +484,412 @@ export function ResolveThisBinding(realm: Realm): NullValue | ObjectValue | Abst
   return envRec.GetThisBinding();
 }
 
-export function BindingInitialization(realm: Realm, node: BabelNode, value: Value, environment: void | LexicalEnvironment) {
+export function BindingInitialization(realm: Realm, node: BabelNodeLVal, value: Value, strictCode: boolean, environment: void | LexicalEnvironment) {
   if (node.type === "ArrayPattern") { // ECMA262 13.3.3.5
     // 1. Let iterator be ? GetIterator(value).
+    let iterator = GetIterator(realm, value);
+
     // 2. Let iteratorRecord be Record {[[Iterator]]: iterator, [[Done]]: false}.
+    let iteratorRecord = {
+      $Iterator: iterator,
+      $Done: false,
+    };
+
     // 3. Let result be IteratorBindingInitialization for ArrayBindingPattern using iteratorRecord and environment as arguments.
+    try {
+      IteratorBindingInitialization(realm, node.elements, iteratorRecord, strictCode, environment);
```

Comment:
`result` isn’t really used and `IteratorBindingInitialization` returns void at the moment. I’ll do it for explicitness anyway 👍